features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp

```diff
@@ -0,0 +1,333 @@
+}
+
+void NRFCordioHCIDriver::do_initialize()
+{
+	if(_is_init) {
+		return;
+	}
+
+	_is_init = true;
+
+    // Setup BB & LL config
+    LlInitRtCfg_t ll_init_cfg =
+    {
+        .pBbRtCfg     = &_bb_cfg,
+        .wlSizeCfg    = 4,
+        .rlSizeCfg    = 4,
+        .plSizeCfg    = 4,
+        .pLlRtCfg     = &_ll_cfg,
+        .pFreeMem     = _stack_buffer,
+        .freeMemAvail = CORDIO_LL_MEMORY_FOOTPRINT
+    };
+
+    // Override RTOS idle thread
+    rtos_attach_idle_hook(idle_hook);
+
+    /* switch to more accurate 16 MHz crystal oscillator (system starts up using 16MHz RC oscillator) */
+    NRF_CLOCK->EVENTS_HFCLKSTARTED = 0;
+    NRF_CLOCK->TASKS_HFCLKSTART    = 1;
+    while (NRF_CLOCK->EVENTS_HFCLKSTARTED == 0)
+    {
+    }
+
+    /* configure low-frequency clock */
+    NRF_CLOCK->LFCLKSRC             = (CLOCK_LFCLKSRC_SRC_Xtal << CLOCK_LFCLKSRC_SRC_Pos);
+    NRF_CLOCK->EVENTS_LFCLKSTARTED  = 0;
+    NRF_CLOCK->TASKS_LFCLKSTART     = 1;
+    while (NRF_CLOCK->EVENTS_LFCLKSTARTED == 0)
+    {
+    }
+    NRF_CLOCK->EVENTS_LFCLKSTARTED = 0;
+
+    // Start RTC0
+    NRF_RTC0->TASKS_STOP  = 1;
+    NRF_RTC0->TASKS_CLEAR = 1;
+    NRF_RTC0->PRESCALER   = 0;            /* clear prescaler */
+    NRF_RTC0->TASKS_START = 1;
+
+    // Cycle radio peripheral power to guarantee known radio state
+    NRF_RADIO->POWER = 0;
+    NRF_RADIO->POWER = 1;
+
+    // For some reason, the mbed target uses this (TIMER0_IRQHandler_v) vector name instead of the "standard" TIMER0_IRQHandler one
+    NVIC_SetVector(TIMER0_IRQn, (uint32_t)TIMER0_IRQHandler);
+
+    // Extremely ugly
+    for(uint32_t irqn = 0; irqn < 32; irqn++)
+    {
+    		uint8_t prio = NVIC_GetPriority((IRQn_Type)irqn);
+    		if( prio < 2 ) {
+    			NVIC_SetPriority((IRQn_Type)irqn, 2);
+    		}
+    }
+
+    // WARNING
+    // If a submodule does not have enough space to allocate its memory from buffer, it will still allocate its memory (and do a buffer overflow) and return 0 (as in 0 byte used)
+    // however that method will still continue which will lead to undefined behaviour
+    // So whenever a change of configuration is done, it's a good idea to set CORDIO_LL_MEMORY_FOOTPRINT to a high value and then reduce accordingly
+    uint32_t mem_used = LlInitControllerExtInit(&ll_init_cfg);
+    if( mem_used < CORDIO_LL_MEMORY_FOOTPRINT )
+    {
+        // Sub-optimal, give warning
+        DBG_WARN("NRFCordioHCIDriver: CORDIO_LL_MEMORY_FOOTPRINT can be reduced to %lu instead of %lu", mem_used, CORDIO_LL_MEMORY_FOOTPRINT);
+    }
+
+    // BD Addr
+    bdAddr_t bd_addr;
+    PlatformLoadBdAddress(bd_addr);
+    LlSetBdAddr((uint8_t *)&bd_addr);
+    LlMathSetSeed((uint32_t *)&bd_addr);
+
+//#ifdef DEBUG
+//    WsfTraceRegister(wsf_trace_handler);
+//#endif
+
+    // We're sharing the host stack's event queue
+}
+
+void NRFCordioHCIDriver::do_terminate()
+{
+
+}
+
+void NRFCordioHCIDriver::start_reset_sequence()
+{
+    // Make sure extended adv is init
+    DmExtAdvInit();
```

Comment:
Any reason to put that here ? 
That function is called by Cordio BLE if the controller supports extended advertising.

Reply:
This needs to be called before the reset sequence to activate the extended reset sequence required to fetch extended advertising related info (number of advertising sets supported, etc)

Reply:
I'm disappointed with the inconsistency of the reset sequence with EA; I suppose we should enable it by default in CordioBLE then if not supported fallback to legacy.

Reply:
Agreed